packages/dds/tree/src/shared-tree/sharedTreeChangeEnricher.ts

```diff
@@ -0,0 +1,91 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { unreachableCase } from "@fluidframework/core-utils/internal";
+import {
+	DeltaDetachedNodeId,
+	DetachedFieldIndex,
+	IEditableForest,
+	RevisionTag,
+	tagChange,
+	visitDelta,
+} from "../core/index.js";
+import {
+	TreeChunk,
+	chunkTree,
+	defaultChunkPolicy,
+	intoDelta,
+	relevantRemovedRoots,
+	updateRefreshers as updateDataChangeRefreshers,
+} from "../feature-libraries/index.js";
+import { disposeSymbol } from "../util/index.js";
+import { ChangeEnricherCheckout } from "./defaultCommitEnricher.js";
+import { updateRefreshers } from "./sharedTreeChangeFamily.js";
+import { SharedTreeChange } from "./sharedTreeChangeTypes.js";
+
+export class SharedTreeChangeEnricher implements ChangeEnricherCheckout<SharedTreeChange> {
+	/**
+	 * @param forest - The state based on which to enrich changes.
+	 * Exclusively owned by the constructed instance.
+	 * @param removedRoots - The set of removed roots based on which to enrich changes.
+	 * Exclusively owned by the constructed instance.
+	 */
+	public constructor(
+		private readonly forest: IEditableForest,
+		private readonly removedRoots: DetachedFieldIndex,
+	) {}
+
+	public updateChangeEnrichments(
+		change: SharedTreeChange,
+		revision: RevisionTag,
+	): SharedTreeChange {
+		const taggedChange = tagChange(change, revision);
+		return updateRefreshers(
+			taggedChange,
+			this.getDetachedRoot,
+			relevantRemovedRoots,
+			updateDataChangeRefreshers,
+		);
+	}
+
+	private readonly getDetachedRoot = (id: DeltaDetachedNodeId): TreeChunk | undefined => {
```

Comment:
Usually when I see people do this it's because the function closes over `this`, but it is also being passed around as a function pointer. If it's a regular method (rather than an arrow function), then the `this` will be wrong when you invoke it in a different context. To fix that, you can make it an arrow function or you can bind it before calling. I think that's probably what's going on here.

Reply:
T'is indeed.

Reply:
There's another option too - you don't need to bind every time, you can do ".call" (or ".apply") instead of ".bind" if you aren't needing to keep the altered function around. I assume that has better performance than repeatedly calling bind, but worse performance than binding once ahead of time (or making it an arrow function as you did here), though I don't actually know what the perf characteristics are.

Reply:
Don't `.call` and `.apply` require passing all parameters though?